### PR TITLE
feat: new onboarding

### PR DESCRIPTION
**Pull request by fbwoolf**
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/hirosystems/stacks-wallet-web/actions/runs/1671852321).

This PR updates onboarding to match the new designs in Figma. It refactors how we use global-styles and the shared Header component. In addition, there is now a `pages.styles.tsx` that handles layout styles for the pages.

#2099 

Screen Shot 2022-01-04 at 5 20 19 PM

Screen Shot 2022-01-04 at 5 20 30 PM

Screen Shot 2022-01-04 at 5 20 51 PM

Screen Shot 2022-01-04 at 5 21 13 PM

Screen Shot 2022-01-04 at 5 21 32 PM

Screen Shot 2022-01-04 at 5 21 46 PM

cc/ @kyranjamie @beguene

### CODE REVIEW

File: src/app/common/utils.ts [hunk 1]
```diff
@@ -275,3 +275,6 @@ type ViewMode = 'full' | 'popup';
 export function getViewMode(): ViewMode {
   return document.location.pathname.startsWith('/index.html') ? 'full' : 'popup';
```
Comment (beguene) [inline src/app/common/utils.ts]: maybe we could reuse the logic below to check if it's FullPage.

File: src/shared/constants.ts [hunk 2]
```diff
@@ -2,6 +2,10 @@ import { ChainID } from '@stacks/transactions';
 
 export const gaiaUrl = 'https://hub.blockstack.org';
 
+export const FULL_PAGE_MAX_WIDTH = 440;
```
Comment (beguene) [inline src/shared/constants.ts]: Maybe we could move the constants related to layouts in its own file.

**Review by beguene — APPROVED** [review]
Nicely done ! 👍

File: src/app/pages/receive-tokens/receive-tokens.tsx [hunk 3]
```diff
@@ -29,7 +31,7 @@ export const ReceiveTokens = () => {
   };
 
   return (
-    <Stack spacing="loose" textAlign="center">
+    <Stack className={isFullPage ? fullPageContent : undefined} spacing="loose" textAlign="center">
```
Comment (kyranjamie) [inline src/app/pages/receive-tokens/receive-tokens.tsx:35]: Does this work? This passes a component to `className` right?

File: src/app/pages/save-secret-key/save-secret-key.styles.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import { css } from '@emotion/css';
+
+import { FULL_PAGE_MAX_WIDTH } from '@shared/constants';
+
+export const fullPageContent = css`
+  align-items: center;
+  height: 100%;
+  justify-content: center;
+  max-width: ${FULL_PAGE_MAX_WIDTH}px;
+  padding: 0px 24px;
```
Comment (kyranjamie) [inline src/app/pages/save-secret-key/save-secret-key.styles.tsx]: This hard codes a padding value, while we have semantic monikers spacing units. How could you avoid repetition of these values?

File: src/app/pages/onboarding/welcome/welcome.layout.tsx [hunk 5]
```diff
@@ -13,26 +23,49 @@ export function WelcomeLayout(props: WelcomeLayoutProps): JSX.Element {
...
-          Sign in with Secret Key
-        </Link>
-      </Stack>
+    <Stack isInline={isFullPage} width="100%">
+      <Flex flexGrow={1} justifyContent="center" order={isFullPage ? 1 : 0}>
+        {isFullPage ? (
+          <img src={WelcomeStacksFull} />
+        ) : (
+          <img src={WelcomeStacksPopup} width="394px" />
+        )}
```
Comment (kyranjamie) [inline src/app/pages/onboarding/welcome/welcome.layout.tsx]: We could make a tidy wrapper for this, either a fn or component: 

```tsx
<WhenPageMode 
  full={<img src={WelcomeStacksFull} />} 
  popup={<img src={WelcomeStacksPopup} width="394px" />}
/>
// or
{whenPageMode({
  full: <img src={WelcomeStacksFull} />,
  popup: <img src={WelcomeStacksPopup} width="394px" />
})}
```

**Review by kyranjamie — APPROVED** [review]
Good work @fbwoolf, some nice tidying too.

I wonder if there are some styles that could be repeated less. There are quite a few repeated props, `height="48px"` is used a load of times, for example. Could we make an `<OnboardingButton />` if this is used so frequently?

**Comment by kyranjamie** [comment]
The `dev` version label should align with the title's baseline 

image

Should we have a max size for the illustration, so it does scale fully on 4k/big monitors?
image

File: src/app/common/utils.ts (hunk 1 above)
Comment (fbwoolf) [reply src/app/common/utils.ts]: Yep, this function is now unused. I have removed it locally, bc of the failing code check.

File: src/app/pages/onboarding/welcome/welcome.layout.tsx (hunk 5 above)
Comment (fbwoolf) [reply src/app/pages/onboarding/welcome/welcome.layout.tsx]: Good idea, will do.

File: src/app/pages/receive-tokens/receive-tokens.tsx (hunk 3 above)
Comment (fbwoolf) [reply src/app/pages/receive-tokens/receive-tokens.tsx:35]: Yep, this works with emotion ...it is a template literal.
https://emotion.sh/docs/@emotion/css#css

File: src/app/pages/save-secret-key/save-secret-key.styles.tsx (hunk 4 above)
Comment (fbwoolf) [reply src/app/pages/save-secret-key/save-secret-key.styles.tsx]: This file was left by mistake. There is only one place now where they are all shared `pages.styles.tsx`. I have removed this file locally, thx.

Comment (fbwoolf) [reply src/app/pages/save-secret-key/save-secret-key.styles.tsx]: Also, the hardcoded padding has been removed.

File: src/shared/constants.ts (hunk 2 above)
Comment (fbwoolf) [reply src/shared/constants.ts]: Ok, yep, that makes sense. 👍

**Comment by fbwoolf** [comment]
> I wonder if there are some styles that could be repeated less. There are quite a few repeated props, `height="48px"` is used a load of times, for example. Could we make an `<OnboardingButton />` if this is used so frequently?

Yep, I wanted to get this PR up but I agree I can simplify things further ...def lots can be done to further tidy up our styling.

**Comment by fbwoolf** [comment]
> The `dev` version label should align with the title's baseline
> 
Good catch, was thinking this was supposed to be centered. Will fix.
> 
> Should we have a max size for the illustration, so it does scale fully on 4k/big monitors?
>
Right now, I don't feel like our styling is very responsive.

**Comment by fbwoolf** [comment]
Screen Shot 2022-01-06 at 7 05 50 PM

Anyone know where `back it up later` is supposed to go? Wouldn't these two options go to the same place?

**Comment by beguene** [comment]
> Screen Shot 2022-01-06 at 7 05 50 PM
> 
> Anyone know where `back it up later` is supposed to go? Wouldn't these two options go to the same place?

Agree, it looks like both should go to the same page. Maybe 'back it up later' means we need additionally store that value so we can later remind the user that he needs to backup.

**Comment by kyranjamie** [comment]
For these initial changes, we should omit it completely. It's dependent on there being all the other updates pushing the user to save it.

**Pull request merged**